extensions/typescript-language-features/src/languageFeatures/refactor.ts

```diff
@@ -516,16 +541,24 @@ class TypeScriptRefactorProvider implements vscode.CodeActionProvider<TsCodeActi
 			return undefined;
 		}
 
-		const actions = Array.from(this.convertApplicableRefactors(document, response.body, rangeOrSelection)).filter(action => {
+		const applicableRefactors = await this.convertApplicableRefactors(document, response.body, rangeOrSelection);
+		const actions = await Promise.all(applicableRefactors.map(async (action) => {
 			if (this.client.apiVersion.lt(API.v430)) {
 				// Don't show 'infer return type' refactoring unless it has been explicitly requested
 				// https://github.com/microsoft/TypeScript/issues/42993
 				if (!context.only && action.kind?.value === 'refactor.rewrite.function.returnType') {
 					return false;
 				}
 			}
+			if (action.kind?.value === Move_NewFile.kind.value) {
+				const navigationTree = await this.client.execute('navtree', { file: document.uri.path }, nulToken);
+				if (navigationTree.type !== 'response') {
+					return;
+				}
+				return MoveToFileCodeAction.isOnDeclarationName(navigationTree.body, rangeOrSelection);
```

Comment:
The move refactors should likely still show if you explicitly request them, such as using the `refactor` context menu or a keybinding. You can check the code action trigger for that